crates/oxc_linter/src/rules/jest/consistent_test_it.rs

```diff
@@ -0,0 +1,916 @@
+            parse_jest_fn_call(call_expr, possible_jest_node, ctx)
+        else {
+            return;
+        };
+
+        if matches!(jest_fn_call.kind, JestFnKind::General(JestGeneralFnKind::Describe)) {
+            let scope_id = node.scope_id();
+            let current_count = describe_nesting_hash.get(&scope_id).unwrap_or(&0);
+            describe_nesting_hash.insert(scope_id, *current_count + 1);
+            return;
+        }
+
+        let is_test = matches!(jest_fn_call.kind, JestFnKind::General(JestGeneralFnKind::Test));
+        let fn_to_str = Self::to_str(&self.within_fn);
+
+        if is_test && describe_nesting_hash.is_empty() && !jest_fn_call.name.ends_with(&fn_to_str) {
+            let opposite_test_keyword = Self::get_opposite_test_case(&self.within_fn);
+            if let Some((span, prefer_test_name)) = Self::get_prefer_test_name_and_span(
+                call_expr.callee.get_inner_expression(),
+                &jest_fn_call.name,
+                &fn_to_str,
+            ) {
+                ctx.diagnostic_with_fix(
+                    consistent_method(plugin_name, &fn_to_str, &opposite_test_keyword, span),
+                    |fixer| fixer.replace(span, prefer_test_name),
+                );
+            }
+        }
+
+        let describe_to_str = Self::to_str(&self.within_describe);
+
+        if is_test
+            && !describe_nesting_hash.is_empty()
+            && !jest_fn_call.name.ends_with(&describe_to_str)
+        {
+            let opposite_test_keyword = Self::get_opposite_test_case(&self.within_describe);
+            if let Some((span, prefer_test_name)) = Self::get_prefer_test_name_and_span(
+                call_expr.callee.get_inner_expression(),
+                &jest_fn_call.name,
+                &describe_to_str,
+            ) {
+                ctx.diagnostic_with_fix(
+                    consistent_method_within_describe(
+                        plugin_name,
+                        &describe_to_str,
+                        &opposite_test_keyword,
+                        span,
+                    ),
+                    |fixer| fixer.replace(span, prefer_test_name),
+                );
+            }
+        }
+    }
+
+    fn to_enum(test_case_name: &str) -> Option<TestCaseName> {
+        match test_case_name {
+            "fit" => Some(TestCaseName::Fit),
+            "it" => Some(TestCaseName::IT),
+            "test" => Some(TestCaseName::Test),
+            "xtest" => Some(TestCaseName::Xtest),
+            "xit" => Some(TestCaseName::Xit),
+            _ => None,
+        }
+    }
+
+    fn to_str(test_name: &TestCaseName) -> String {
+        let test_case_name = match test_name {
+            TestCaseName::Fit => "fit",
+            TestCaseName::IT => "it",
+            TestCaseName::Test => "test",
+            TestCaseName::Xtest => "xtest",
+            TestCaseName::Xit => "xit",
+        };
+
+        test_case_name.to_string()
+    }
```

Comment:
Try to implement `std::fmt::Display` for `TestCaseName`

Reply:
Done.